### ISSUE TITLE
Parameters for Plant Genomes

### ISSUE THREAD

**Issue by malonge** [chanzuckerberg/shasta#65]
Hi there,

This is somewhat of a follow up to #54. I tried assembling some tomato data with default parameters and it didn't quite work out. The assembly should be between 750M to 800M, but it was closer to 500M.

I bring it up as a potentially good test case, if you are interested, since the genome is relatively simple (especially since it is very homozygous) and there is test data on SRA for 3 different accessions. The data is associated with [my preprint](https://www.biorxiv.org/content/10.1101/519637v1).

The "BGV" (BGV006775) accession probably has the highest quality data, but the "M82" accession would be most convenient for my testing purposes. 

Thanks

**Comment by paoloczi** [comment]
Thank you Michael - I will definitely give this a try soon and will get back to you by adding comments to this issue.

**Comment by paoloczi** [comment]
I believe you are interested in the data sets in this directory:

http://labshare.cshl.edu/shares/schatzlab/www-data/ragoo/Tomato/M82/ONT_Reads/

I see several data sets there. Can you clarify which ones you are interested in exploring?

**Comment by malonge** [comment]
Hi there,

Yes you can use the data in "ONT_Reads". The same data is also on SRA if that is easier. 

It would be quite nice to see how those M82 reads assemble. You can just throw them all together for the input. M82 has some minion and some gridion reads.

If you go up a few parent directories, you can also find data for BGV. Those reads are of higher quality. 

Thanks

**Comment by paoloczi** [comment]
I did some experimentation with your M82 data set. These reads are shorter and much less accurate that the ones in [the paper](https://www.biorxiv.org/content/10.1101/715722v1), for which default Shasta parameters are optimized.

After experimenting with various assembly parameters, I was able to assemble 769 Mb of sequence, but the N50 for assembled segments was only 414 Kb, versus the 20 Mb or so we got in the paper. 

Here are the assembly parameters I used for that assembly:

--Kmers.k 8 
--MinHash.version 1 
--MinHash.m 3 
--MinHash.minHashIterationCount 200 
--MinHash.minBucketSize 3 
--MinHash.maxBucketSize 30 
--MinHash.minFrequency 3 
--Reads.minReadLength 6000

Note this requires the latest Shasta code on GitHub, as some of those options were not available in release 0.3.0. Until the next release becomes available, you can download a current test build by following the directions [here](https://chanzuckerberg.github.io/shasta/BuildingFromSource.html#DownloadTestBuild).

I posted the assembly for you as follows - however these are temporary links, so if you are interested in looking at this assembly please copy it over soon:

https://s3.us-east-2.amazonaws.com/czi.paolo-tmp1/M82/run11/AssemblySummary.html
https://s3.us-east-2.amazonaws.com/czi.paolo-tmp1/M82/run11/Assembly.fasta
https://s3.us-east-2.amazonaws.com/czi.paolo-tmp1/M82/run11/Assembly.gfa
https://s3.us-east-2.amazonaws.com/czi.paolo-tmp1/M82/run11/Assembly-BothStrands.gfa

I did not analyze assembly quality. Some features in the gfa output, displayed below using [Bandage](https://rrwick.github.io/Bandage/), suggest the presence of highly heterozygous regions, but this could be an artifact of the assembly parameters used.

In summary, I was not able to find a set of assembly parameters that would give a satisfactory assembly for this lower quality data set.

I will also take a look at your BGV dataset.

![image](https://user-images.githubusercontent.com/25391437/67125141-f5abfd80-f1a8-11e9-93aa-83ea01d7d4f5.png)

**Comment by malonge** [comment]
Hi there,

Thank you so much for taking the time to test on this dataset. Indeed, this particular set of reads is on the lower end of the quality scale for us. Most of our accessions, though not yet published, are from the Promethion. 

But hopefully, given these parameters and better data, we can start to see positive results.

Thanks,
Mike

**Comment by malonge** [comment]
I will also do some simple assembly QC and let you know if anything looks very wrong.

**Comment by paoloczi** [comment]
What base caller did you use for the BGV data? Guppy? Which version? We have Bayesian models for repeat counts for two different Guppy versions and I want to pick the one closest to the base caller version you used.

**Comment by malonge** [comment]
Should be Albacore. I will double-check and get the version number.

**Comment by paoloczi** [comment]
I hear that Albacore is obsolete at this point and superseded by Guppy, which delivers significantly better read quality for the same signal. If you still have the signal files, you should consider redoing the base calling using the current Guppy version.

**Comment by malonge** [comment]
Yes we currently use Guppy, but these data were generated over a year ago.

**Comment by malonge** [comment]
Sorry that was an accidental close

**Comment by paoloczi** [comment]
For the BGV dataset, default Shasta parameters don't work well because many alignments have long "holes", hundreds of bases long, where one of the two reads has a high error rate. 

I was able to obtain a reasonable assembly using the latest code on GitHub (commit b54605cb734b550a995092446cea80cae5942ede) and the command line options pasted below. Note release 0.3.0 lacks some of those options.

The assembly includes 794 Mb of sequence with an N50 of 2.4 Mb. This is still about an order of magnitude shorter than what we are getting in our paper, possibly due to the fact that you used Albacore for your base calling, rather than the newer Guppy we used in our paper. 

The assembly is posted at these ***temporary*** links:

https://s3.us-east-2.amazonaws.com/czi.paolo-tmp1/BGV/run3/Assembly.fasta
https://s3.us-east-2.amazonaws.com/czi.paolo-tmp1/BGV/run3/Assembly.gfa
https://s3.us-east-2.amazonaws.com/czi.paolo-tmp1/BGV/run3/Assembly-BothStrands.gfa
https://s3.us-east-2.amazonaws.com/czi.paolo-tmp1/BGV/run3/AssemblySummary.html

Command line options used for this run:

--Align.maxSkip 50  
--MinHash.m 3 
--MinHash.minBucketSize 3
--MinHash.maxBucketSize 30 
--MinHash.minHashIterationCount 30 
--MinHash.minFrequency 3

**Comment by malonge** [comment]
Thanks again so much! Those stats are still quite respectable. And just as with M82, these parameters combined with better data could yield even better results. Thanks for taking the time to look into it.

**Comment by paoloczi** [comment]
Since no additional comments came up, I am closing this issue.
